frontend/packages/console-shared/src/components/namespace/NamespaceMenuToggle.tsx

```diff
@@ -0,0 +1,91 @@
+import * as React from 'react';
+import { MenuToggle, Popper } from '@patternfly/react-core';
+
+const NamespaceMenuToggle = (props: {
+  disabled: boolean;
+  menu: React.ReactElement;
+  menuRef: React.RefObject<HTMLElement>;
+  isOpen: boolean;
+  shortCut?: string;
+  title: string;
+  onToggle: (state: boolean) => void;
+}) => {
+  const { menu, isOpen, shortCut, title, onToggle, disabled, menuRef } = props;
+
+  const toggleRef = React.useRef(null);
+  const containerRef = React.useRef(null);
+
+  const handleMenuKeys = (event) => {
+    if (
+      shortCut &&
+      event.key === shortCut &&
+      event.target.nodeName !== 'INPUT' &&
+      event.target.nodeName !== 'TEXTAREA'
+    ) {
+      onToggle(true);
+      event.stopPropagation();
+      event.preventDefault();
+    }
+
+    if (menuRef.current) {
+      if (event.key === 'Escape') {
+        onToggle(false);
+        toggleRef.current.focus();
+      }
+      if (!menuRef.current?.contains(event.target) && event.key === 'Tab') {
+        onToggle(false);
+      }
+    }
+  };
+
+  const handleMenuClick = (event) => {
+    if (
+      menuRef.current &&
+      !menuRef.current?.contains(event.target) &&
+      // Checking to see if user clicked on a favorite icon.  This is needed because
+      // if unfavoriting a item, PF removes the item from the DOM before
+      // the click event is registered
+      !event.target.closest('.pf-m-favorite') &&
```

Comment:
It appears that when clicking on the "unfavorite" star for an item in the "Favorites" section is removed from the DOM before the click event handler is triggered.  This was causing `!menuRef.current?.contains(event.target) ` to return true and the menu would close when it shouldn't.

Because the item is no longer in the menu DOM, I was left with checking by classname which is less than ideal.